core/arch/arm/kernel/unwind_arm32.c

```diff
@@ -130,6 +130,7 @@ static struct unwind_idx *find_index(uint32_t addr, vaddr_t exidx,
 	min = 0;
 	max = (idx_end - idx_start) / sizeof(struct unwind_idx);
 
+	addr -= 2;
```

Comment:
Perhaps we should call this function as:
``` C
/*
 * Find the item to run. Subtract 2 from PC to make sure that we're still inside
 * the calling function in case a __no_return function (typically panic()) is
 * called unconditionally and may cause LR and thus this PC to point into the
 * next and entirely unrelated function.
 */
index = find_index(state->start_pc - 2, exidx, exidx_sz);
```
below in `unwind_stack_arm32()` instead.

Reply:
ok, thanks for the comment.

I will push a new version soon.